MagicBots/Overseer/Framework/Discord/OverseerHostedService.cs

```diff
@@ -1,23 +1,38 @@
 ﻿using Discord;
 using Discord.WebSocket;
+using MagicBots.Overseer.Framework.Services;
 using Microsoft.Extensions.Configuration;
 using Microsoft.Extensions.Logging;
-using System;
-using System.Collections.Generic;
-using System.Linq;
-using System.Text;
 using System.Threading.Tasks;
 
 namespace MagicBots.Overseer.Framework.Discord
 {
     public class OverseerHostedService : DiscordClientHostedService
     {
-        public OverseerHostedService(ILogger logger, IConfiguration configuration) : base(logger, configuration)
+        private readonly TriggeringService _triggeringService;
+
+        public OverseerHostedService(ILogger logger, IConfiguration configuration,
+            TriggeringService triggeringService) : base(logger, configuration)
         {
+            _triggeringService = triggeringService;
+
+            Client!.MessageReceived += MessageReceivedAsync;
```

Comment:
I think this will throw because Client can be null, right?

Reply:
Also, my intent was to register all of these under ConfigureClientAsync, but I guess this does work too. Which way do you think is better?

Reply:
Ok, in theory you should be able to rely on the null checking behavior from dotnet. I'm not 100% convinced that it's actually working, but it's supposed to prevent you from calling ConfigureClientAsync with a null client in the first place unless you changed the signature to `DiscordSocketClient? client`